app-android/src/main/java/com/ustadmobile/port/android/view/ContentEntryEdit2Fragment.kt

```diff
@@ -459,4 +479,233 @@ class ContentEntryEdit2Fragment(
+
+            Spacer(modifier = Modifier.height(15.dp))
+
+            Text(updateContentText)
+        }
+
+        Spacer(modifier = Modifier.height(15.dp))
+
+        if (uiState.entity?.leaf == true){
+            Text(text = stringResource(id = R.string.supported_files))
+        }
+
+        Spacer(modifier = Modifier.height(15.dp))
+
+        UstadTextEditField(
+            value = uiState.entity?.title ?: "",
+            label = stringResource(id = R.string.title),
+            error = uiState.titleError,
+            enabled = uiState.fieldsEnabled,
+            onValueChange = {
+                onContentChanged(uiState.entity?.shallowCopy {
+                        title = it
+                    }
+                )
+            }
+        )
+
+        Spacer(modifier = Modifier.height(15.dp))
+
+        UstadTextEditField(
+            value = uiState.entity?.description ?: "",
+            label = stringResource(id = R.string.description),
+            enabled = uiState.fieldsEnabled,
+            onValueChange = {
+                onContentChanged(uiState.entity?.shallowCopy {
+                        description = it
+                    }
+                )
+            }
+        )
+
+        Spacer(modifier = Modifier.height(15.dp))
+
+        CourseBlockEdit(
+            uiState = courseBlockEditUiState,
+            onCourseBlockChange = onCourseBlockChange
+        )
+
+        Spacer(modifier = Modifier.height(15.dp))
+
+        UstadTextEditField(
+            value = uiState.entity?.author ?: "",
+            label = stringResource(id = R.string.entry_details_author),
+            enabled = uiState.fieldsEnabled,
+            onValueChange = {
+                onContentChanged(uiState.entity?.shallowCopy {
+                        author = it
+                    }
+                )
+            }
+        )
+
+        Spacer(modifier = Modifier.height(15.dp))
+
+        UstadTextEditField(
+            value = uiState.entity?.publisher ?: "",
+            label = stringResource(id = R.string.entry_details_publisher),
+            enabled = uiState.fieldsEnabled,
+            onValueChange = {
+                onContentChanged(uiState.entity?.shallowCopy {
+                        publisher = it
+                    }
+                )
+            }
+        )
+
+        Spacer(modifier = Modifier.height(15.dp))
+
+        UstadMessageIdOptionExposedDropDownMenuField(
+            value = uiState.entity?.licenseType ?: 0,
+            options = LicenceConstants.LICENSE_MESSAGE_IDS,
+            label = stringResource(id = R.string.licence),
+            enabled = uiState.fieldsEnabled,
+            onOptionSelected = {
+                onContentChanged(uiState.entity?.shallowCopy {
+                        licenseType = it.value
+                    }
+                )
+            }
+        )
+
+        Spacer(modifier = Modifier.height(15.dp))
+
+        if (uiState.containerStorageOptionVisible){
```

Comment:
Add selectedContainerStorageDir as a variable on UiState

Can use:

UstadExposedDropDownMenuField(
        value = uiState.selectedContainerStorageDir,
        label = stringResource(R.string..),
        options = uiState.containerStorageDirs,
        onOptionSelected = {
            onSelectContainerStorageDir(it)
        }
        itemText = { it.name ?: "" },
        enabled = uiState.fieldsEnabled,
    )